packages/starlight/__tests__/basics/slugs.test.ts

```diff
@@ -43,6 +43,9 @@ describe('slugToParam', () => {
 		expect(slugToParam('dir/page')).toBe('dir/page');
 		expect(slugToParam('dir/sub-dir/page')).toBe('dir/sub-dir/page');
 	});
+	test('normalizes unnormalized strings', () => {
+		expect(slugToParam('اللُّغَةُ-الْعَرَبِيَّةُ')).toBe('اللُّغَةُ-الْعَرَبِيَّةُ');
```

Comment:
Should we provide a bit more details, e.g. for someone without context reading the test thinking the input and output are identical and should match? Maybe something like?

```suggestion
		// The input and output contain visually indistinguishable glyphs with different code point
		// representation that are canonically equivalent that should be normalized using NFC
		// normalization matching Astro's behavior.
		expect(slugToParam('اللُّغَةُ-الْعَرَبِيَّةُ')).toBe('اللُّغَةُ-الْعَرَبِيَّةُ');
```